docs/react/your-first-app/2-taking-photos.md

```diff
@@ -4,143 +4,265 @@ sidebar_label: Taking Photos
+      resultType: CameraResultType.Uri,
+      source: CameraSource.Camera,
+      quality: 100,
+    });
 
-Next, let's expose the photos array from our hook. Update the return statement to include the photos:
+    // CHANGE: Create the `fileName` with current timestamp.
+    const fileName = Date.now() + '.jpeg';
+    // CHANGE: Create `savedImageFile` matching `UserPhoto` interface.
+    const savedImageFile = [
+      {
+        filepath: fileName,
+        webviewPath: capturedPhoto.webPath,
+      },
+      ...photos,
+    ];
+
+    // CHANGE: Update the `photos` array with the new photo.
+    setPhotos(savedImageFile);
+  };
 
-```tsx
-return {
-  photos,
-  takePhoto,
-};
+  return {
+    addNewToGallery,
+    // CHANGE: Update return statement to include `photos` array.
+    photos,
+  };
+}
 ```
 
-And back in the Tab2 component, get access to the photos:
+`usePhotoGallery.ts` should now look like this:
 
-```tsx
-const { photos, takePhoto } = usePhotoGallery();
+```ts
+import { useState, useEffect } from 'react';
+import { Camera, CameraResultType, CameraSource, Photo } from '@capacitor/camera';
+import { Filesystem, Directory } from '@capacitor/filesystem';
+import { Preferences } from '@capacitor/preferences';
+
+export function usePhotoGallery() {
+  const [photos, setPhotos] = useState<UserPhoto[]>([]);
+
+  const addNewToGallery = async () => {
+    // Take a photo
+    const capturedPhoto = await Camera.getPhoto({
+      resultType: CameraResultType.Uri,
+      source: CameraSource.Camera,
+      quality: 100,
+    });
+
+    const fileName = Date.now() + '.jpeg';
+    const savedImageFile = [
+      {
+        filepath: fileName,
+        webviewPath: capturedPhoto.webPath,
+      },
+      ...photos,
+    ];
+
+    setPhotos(savedImageFile);
+  };
+
+  return {
+    addNewToGallery,
+    photos,
+  };
+}
+
+export interface UserPhoto {
+  filepath: string;
+  webviewPath?: string;
+}
 ```
 
-With the photo(s) stored into the main array we can display the images on the screen. Add a [Grid component](https://ionicframework.com/docs/api/grid) so that each photo will display nicely as photos are added to the gallery, and loop through each photo in the Photos array, adding an Image component (`<IonImg>`) for each. Point the `src` (source) to the photo’s path:
+Next, switch to `Tab2.tsx` to display the images. We'll add a [Grid component](../../api/grid.md) to ensure the photos display neatly as they're added to the gallery. Inside the grid, loop through each photo in the `UserPhoto`'s `photos` array. For each item, add an [Image component](../../api/img.md) and set its `src` property to the photo's path.
 
 ```tsx
-<IonContent>
-  <IonGrid>
-    <IonRow>
-      {photos.map((photo, index) => (
-        <IonCol size="6" key={photo.filepath}>
-          <IonImg src={photo.webviewPath} />
-        </IonCol>
-      ))}
-    </IonRow>
-  </IonGrid>
-   <!-- <IonFab> markup  -->
-</IonContent>
+const Tab2: React.FC = () => {
```

Comment:
I think this code would be a good place to include the imports it needs and also include the export for easy copy/pasting:

```
export default Tab2;
```